### PR TITLE
[TMVA][SOFIE] Fix concat operator

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR fixes the Concat operator for concatenating on the 0th axis and adds corresponding test.

### CODE REVIEW

File: tmva/sofie/inc/TMVA/ROperator_Concat.hxx [hunk 1]
```diff
@@ -114,8 +114,10 @@
             out<<"\n//--------- Concat\n";
             // special case for 0 axis that memory is contigous
             if (fAxis == 0) {
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_Concat.hxx:116]: I think we can also use the condition of fAxis=0 if  the input shapes have all one's before. 
We can check if the product of the input shape before fAxis is =1 , i.e.
```
bool hasShapeOnes = true;
for (int i = 0; i < fAxis; i++) {
      if (fInputShapes[0][i] != 1) hasShapeOnes = false;
}
if if (fAxis == 0 || hasShapeOnes ) { 
.... // special case that memory is contigous
.....
```

Comment (sanjibansg) [reply tmva/sofie/inc/TMVA/ROperator_Concat.hxx:116]: Ahh yes, good catch, I will make that change, thanks!

File: tmva/sofie/inc/TMVA/ROperator_Concat.hxx [hunk 2]
```diff
@@ -112,10 +112,17 @@
             }
             std::stringstream out;
             out<<"\n//--------- Concat\n";
-            // special case for 0 axis that memory is contigous
-            if (fAxis == 0) {
-               for(auto &it : fInputs){
-                  out<<SP<<"tensor_" << fOutput<<".insert("<<"tensor_" << fOutput<<".end(),tensor_" << it<<".begin(),tensor_"<<it<<".end());\n";
+            // special case when memory is contigous
+            bool hasShapeOnes = true;
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_Concat.hxx:116]: I think here the loop should be on the tensor dimensions and not on the inputs. All inputs have by definition the same shape, no need to loop on them.
```
for(size_t i = 0; i<fAxis; ++i){
                   if(fInputShapes[0][i]!=1)
                   hasShapeOnes = false;
                   break;  // no need to continue the loop
  }

**Review by lmoneta — CHANGES_REQUESTED** [review]
See my comment on the check for having contigous memory, thanks @sanjibansg if you can correct this

File: tmva/sofie/inc/TMVA/ROperator_Concat.hxx (hunk 2 above)
Comment (sanjibansg) [reply tmva/sofie/inc/TMVA/ROperator_Concat.hxx:116]: yes, I agree. Made the change, thanks!

**Review by lmoneta — APPROVED** [review]
LGTM! 
Thank you Sanjban for this fix!

**Pull request merged**